File: code/go/2021/sept/maximum-number-of-balloons.go
package sept

func maxNumberOfBalloons(text string) int {
	var arr [26]int
	for _, v := range text {
		arr[v-'a']++
	}
	res := arr[1]
	res = min(res, arr[0])
	res = min(res, arr[11]/2)
	res = min(res, arr[14]/2)
	res = min(res, arr[13])

	return res
}

File: archive/python/2021/sept/maximum-number-of-balloons.py
from typing import Collection
import collections

class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        cnt = Collection.Counter(text)
        cntBalloon = collections.Counter("balloon")
        return min([cnt[c] // cntBalloon[c] for c in cntBalloon])
